IOPool/Output/src/RootOutputFile.cc

```diff
@@ -145,19 +155,16 @@ namespace edm {
     runTree_.addAuxiliary<RunAuxiliary>(
         BranchTypeToAuxiliaryBranchName(InRun), pRunAux_, om_->auxItems()[InRun].basketSize_);
 
-    treePointers_[InEvent] = &eventTree_;
-    treePointers_[InLumi] = &lumiTree_;
-    treePointers_[InRun] = &runTree_;
-    treePointers_[InProcess] = nullptr;
+    treePointers_.emplace_back(&eventTree_);
+    treePointers_.emplace_back(&lumiTree_);
+    treePointers_.emplace_back(&runTree_);
+    for (auto& processBlockTree : processBlockTrees_) {
+      treePointers_.emplace_back(processBlockTree.get());
+    }
```

Comment:
Yes. That is the plan I am working towards. One of the biggest reasons I submitted this PR for review at this preliminary time was wondering if you and Chris thought that was reasonable.

Reply:
For example, say process HLT produced products A, B, and C, and a later process PROD produced products D, E, and F. We have two choices:

1. Only 1 TTree. The ProcessBlock entries originally created in HLT processes will only contain products A, B, and C. The entries in those branches for D, E, and F will always contain dummy empty entries. The ProcessBlock entries originally created in PROD processes will always have dummy empty entries for branches for products A, B, and C and only contain real products for D, E, and F.

2. 1 TTree per process. The HLT TTree will only have branches for A, B, and C. The PROD TTree will only have branches for products D, E, and F. I chose that because it seemed more efficient for the persistent format.

At the same time I was planning to have a principal which includes product resolvers for all ProcessBlock products. There is the same choice for principals, but in that case the efficiency issues seem less important and the complexities of having a Principal per process name worse.

Reply:
Thanks, I see. Makes sense, even if it would deviate from how Event/Lumi/Run trees (on the other hand ProcessBlock is very different concept).

Reply:
Will these ProcessBlock TTrees ever have more than 1 entry?

Reply:
Yes. After file merging has occurred they may have multiple entries. If no file merging occurs, they will be one entry only.

Reply:
If the different processes could have different number of entries then it does seem to make sense to have a TTree per process.

Reply:
They could have a different number of entries.

A couple other comments. If all products get dropped for a particular process, then the corresponding TTree is not created at all. Also, I was not planning on having any Auxiliary class for ProcessBlock TTrees.